crates/forge-runner/src/sierra_casm_runner.rs

```diff
@@ -0,0 +1,549 @@
+use blockifier::execution::syscalls::hint_processor::SyscallHintProcessor;
+use std::collections::HashMap;
+
+use crate::test_execution_syscall_handler::TestExecutionSyscallHandler;
+use cairo_felt::Felt252;
+use cairo_lang_casm::instructions::Instruction;
+use cairo_lang_casm::{casm, casm_extend};
+use cairo_lang_runner::casm_run::RunFunctionContext;
+pub use cairo_lang_runner::casm_run::{CairoHintProcessor, StarknetState};
+use cairo_lang_runner::{token_gas_cost, Arg, RunResult, RunResultValue, RunnerError};
+use cairo_lang_sierra::extensions::bitwise::BitwiseType;
+use cairo_lang_sierra::extensions::core::{CoreLibfunc, CoreType};
+use cairo_lang_sierra::extensions::ec::EcOpType;
+use cairo_lang_sierra::extensions::enm::EnumType;
+use cairo_lang_sierra::extensions::gas::{CostTokenType, GasBuiltinType};
+use cairo_lang_sierra::extensions::pedersen::PedersenType;
+use cairo_lang_sierra::extensions::poseidon::PoseidonType;
+use cairo_lang_sierra::extensions::range_check::RangeCheckType;
+use cairo_lang_sierra::extensions::segment_arena::SegmentArenaType;
+use cairo_lang_sierra::extensions::starknet::syscalls::SystemType;
+use cairo_lang_sierra::extensions::{ConcreteType, NamedType};
+use cairo_lang_sierra::program::{Function, GenericArg};
+use cairo_lang_sierra::program_registry::ProgramRegistry;
+use cairo_lang_sierra_ap_change::calc_ap_changes;
+use cairo_lang_sierra_gas::gas_info::GasInfo;
+use cairo_lang_sierra_to_casm::compiler::CairoProgram;
+use cairo_lang_sierra_to_casm::metadata::{
+    calc_metadata, Metadata, MetadataComputationConfig, MetadataError,
+};
+use cairo_lang_sierra_type_size::{get_type_size_map, TypeSizeMap};
+use cairo_lang_starknet::contract::ContractInfo;
+use cairo_lang_utils::casts::IntoOrPanic;
+use cairo_lang_utils::extract_matches;
+use cairo_lang_utils::ordered_hash_map::OrderedHashMap;
+use cairo_vm::serde::deserialize_program::{BuiltinName, HintParams, ReferenceManager};
+use cairo_vm::types::program::Program;
+use cairo_vm::types::relocatable::MaybeRelocatable;
+use cairo_vm::vm::errors::cairo_run_errors::CairoRunError;
+use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
+use cairo_vm::vm::runners::cairo_runner::CairoRunner;
+use cairo_vm::vm::vm_core::VirtualMachine;
+use num_traits::ToPrimitive;
+
+// almost entirely copied from cairo_lang_runner
+// fields are `pub` now
```

Comment:
Has anything else changed?

Reply:
yup, there is a marked code below
`finalize` function and it's usage

Reply:
Ok, let's maybe write that in this comment at the top